sqlmesh/core/loader.py

```diff
@@ -264,7 +265,14 @@ def _glob_paths(
 
         for filepath in path.glob(f"**/*{extension}"):
             for ignore_pattern in ignore_patterns:
-                if filepath.match(ignore_pattern):
+                # We try to match both ignore_pattern itself and every file returned by glob,
+                # so that we will always ignore file names that do not appear in the latter.
+                if filepath.match(ignore_pattern) or any(
```

Comment:
is th ere any performance problem here?

Reply:
I haven't benchmarked this, but based on their [docs](https://docs.python.org/3/library/glob.html#glob.glob):

> If recursive is true, the pattern “**” will match any files and zero or more directories, subdirectories and symbolic links to directories. If the pattern is followed by an [os.sep](https://docs.python.org/3/library/os.html#os.sep) or [os.altsep](https://docs.python.org/3/library/os.html#os.altsep) then files will not match.

Given that `**` patterns weren't working until now, I believe it's fair to assume that existing `ignore_patterns` lists only contain absolute/relative file paths or patterns with a single `*` in them. My intuition is that `glob`ing those patterns shouldn't incur any significant performance hits compared to the existing  `filename.match(...)` call.